04.ls/ls.rb

```diff
@@ -1,19 +1,114 @@
 # frozen_string_literal: true
 
 require 'optparse'
+require 'ffi-xattr'
+require 'etc'
+
 COLUMN_COUNT = 3
 SPACE = 1
+PERMISSION_TYPE = ['--x', '-w-', '-wx', 'r--', 'r-x', 'rw-', 'rwx'].freeze
 
 def main
   options = {}
   opt = OptionParser.new
+  opt.on('-r') { |v| v }
   opt.on('-a') { |v| v }
+  opt.on('-l') { |v| v }
   opt.parse!(ARGV, into: options)
 
   path =  "#{Dir.getwd}/"
   entries = Dir.entries(path).sort
-  filtered_entries = options.key?(:a) ? entries : entries.reject { |entry| entry.start_with?('.') }
-  entries_with_suffix = append_suffix_by_file_type(filtered_entries, path)
+
+  sorted_entries = options.key?(:r) ? entries.reverse : entries
+  filtered_entries = options.key?(:a) ? sorted_entries : sorted_entries.reject { |entry| entry.start_with?('.') }
+  options.key?(:l) ? show_long_format(filtered_entries, path) : show_short_format(filtered_entries, path)
+end
+
+def show_long_format(entries, path)
+  entries_long_format = []
+  total_blocks = 0
+
+  entries.each do |entry_name|
+    entry_details = File.lstat(entry_name)
```

Comment:
```suggestion
    lstat = File.lstat(entry_name)
```

とした方がデータの中身がわかりやすいと思います。

https://bootcamp.fjord.jp/pages/readable-variable-name